activerecord/lib/active_record/asynchronous_queries_tracker.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  class AsynchronousQueriesTracker # :nodoc:
+    class Session # :nodoc:
+      def initialize
+        @active = true
+      end
+
+      def active?
+        @active
+      end
+
+      def finalize
```

Comment:
I don't know Rails' code style conventions, but should we add a bang here to indicate mutation?

```suggestion
      def finalize!
```

Reply:
No we don't add bangs to indicate mutation.